src/models/Channel.js

```diff
@@ -10,7 +10,7 @@ const ChannelSchema = new mongoose.Schema({
   users: [
     {
       type: mongoose.Schema.Types.ObjectId,
-      ref: User,
+      ref: 'user',
     },
   ],
```

Comment:
Thinking about this, in the future, if we want the user to be removed from a channel we will need to remove the user from the channel and remove the channel from the user.  Just curious, if there is a different approach for this that could be better but I'm not 100% sure.  This works but I just wanted to share my thoughts while I'm looking at this.

Reply:
The stackoverflow question @Igor333m shared yesterday deals with exactly this. From what I understood, we would create another collection that maps the relationships between users and channels. And instead of pushing new users to channel and channel to users, we add to this collection. [Worth a read](https://stackoverflow.com/questions/46019149/many-to-many-with-mongoose/46020968#46020968)
```
const UserChannelSchema = new Schema({
  user_Id: { type: Schema.Types.ObjectId, ref: 'user', required: true },
  channel_Id: { type: Schema.Types.ObjectId, ref: 'channel', required: true }
});
```